pulpcore/cli/common/generic.py

```diff
@@ -805,6 +805,13 @@ def _type_callback(ctx: click.Context, param: click.Parameter, value: t.Optional
     expose_value=False,
 )
 
+repositories_option = click.option(
```

Comment:
It might be worth adding the callback (e.g., `get_list_of_repos` returning a list of contexts) here so it can be used by other commands in the future.

Reply:
It should be a resource option in the first place.
Also i am beginning to think that there is no nice way to interface this feature.

Reply:
@mdellweg I also think so. I initially though that working with multiple repositories is quite a generic thing, but later found out it actually isn't, so it's quite hard to think of a nice interface that doesn't break the style of the other commands.

Reply:
Multiple repositories is not the issue. See https://github.com/pulp/pulp-cli/blob/main/pulpcore/cli/core/exporter.py#L26
Multiple versions is a bit harder: https://github.com/pulp/pulp-cli/blob/main/pulpcore/cli/core/export.py#L131

The issue is an api endpoint that is supposed to "Work on these repositories, but don't touch these versions.".

Reply:
I made selecting repositories a resource option now, the only difference now is that you have to type (example) "file:file:my_repo" instead of "my_repo", which may slow down the experience, but I actually like it for differentiating between different plugins' repositories. 

When it comes to the keeplisted versions, I let it be as a simple list of hrefs. I can make it more elegant if it's really needed (e.g. entering the version as an integer), but otherwise I'm for keeping it this simple as I believe the option is not used that much often (I may be wrong tho).

Reply:
Have you seen this? https://github.com/pulp/pulp-cli/blob/main/tests/scripts/pulpcore/test_pulpexporter.sh#L58
Consistency is key.